### PR TITLE
More Detailed WAVE Template

### PR DESCRIPTION
**Pull request by iluvcapra**
This version is more structured to make it easier to parse more kinds of chunks, and it handles RF64 files.

### CODE REVIEW

**Comment by kainjow** [comment]
FYI @armadsen

File: templates/Audio/WAV.tcl [hunk 1]
```diff
@@ -1,38 +1,224 @@
...
+little_endian
+requires 8 "57 41 56 45" ;# WAVE
+set header_signature [ascii 4 "Header Signature"]
+set riff_size [uint32 "RIFF Size"]
+set riff_form [ascii 4 "RIFF Form"]
+if {$header_signature == "RF64" || $header_signature == "BW64"} {
+    parse_rf64 
+} elseif {$header_signature == "RIFF" } {
+    parse_wave $riff_size
+}
```
Comment (kainjow) [inline templates/Audio/WAV.tcl:224]: nit: add ending new line.

I also wonder if it should error out here in an ending `else` condition if the signature is unsupported? Can we use the [error](https://www.tcl.tk/man/tcl8.6/TclCmd/error.htm) command and have that message show in HF (which should handle generic script errors)?

Comment (iluvcapra) [reply templates/Audio/WAV.tcl]: The requested newline has been added.

An `error` command would be useful in many places, particularly if it captured the current file pos and reported it.

Comment (kainjow) [reply templates/Audio/WAV.tcl]: Are you able to try out the existing TCL error command I linked above?

**Pull request merged**